dagmc_stats/dagmc_stats.py

```diff
@@ -471,12 +474,88 @@ def get_roughness(my_core, native_ranges):
 
     outputs
     -------
-    roughness : (numpy array) the roughness for all surfaces in the meshset
+    roughness : (dictionary) the roughness for all vertices in the meshset
+    stored in the form of vert : local roughness value
     """
     tri_vert_data, all_verts = get_tri_vert_data(my_core,
                                                     native_ranges[types.MBTRI])
     gc_all = get_gaussian_curvature(my_core, all_verts, tri_vert_data)
-    roughness = np.zeros(len(all_verts))
-    for vert_idx, vert_i in enumerate(all_verts):
-        roughness[vert_idx] = get_lri(vert_i, gc_all, tri_vert_data, my_core)
+    roughness = {}
+    for vert_i in all_verts:
+        roughness[vert_i] = get_lri(vert_i, gc_all, tri_vert_data, my_core)
     return roughness
+    
+
+def get_tri_roughness(my_core, native_ranges, roughness):
+    """Get triangle average roughness
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    native_ranges : a dictionary containing ranges for each native type
+    in the file (VERTEX, TRIANGLE, ENTITYSET)
+    roughness : (dictionary) the roughness for all vertices in the meshset
+    stored in the form of vert : local roughness value
+
+    outputs
+    -------
+    tri_roughness : (dictionary) the avergae roughness values for all triangles
+    in the meshset stored in the form of tri : average roughness value
+    """
+    tri_roughness = {}
+    for tri in native_ranges[types.MBTRI]:
+        three_verts = list(my_core.get_adjacencies(tri, 0, op_type=1))
+        sum_lr = 0
+        for vert in three_verts:
+            sum_lr += roughness[vert]
+        tri_roughness[tri] = sum_lr/3
+    return tri_roughness
+    
+    
+def add_tag(my_core, tag_name, tag_dic, tag_type):
+    """Add tag according to given tag information
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    tag_name : tag name
+    tag_dic : a dictionary containing eh:data
+    tag_type : tag type
+    """
+    # create the tag handle
+    tag_eh = \
+        my_core.tag_get_handle(tag_name, size=1,
+                                tag_type=tag_type,
+                                storage_type=types.MB_TAG_SPARSE,
+                                create_if_missing=True)
+    for eh, data in tag_dic.items():
+        # assign data to the tag:
+        my_core.tag_set_data(tag_eh, eh, data)
+
+
+def avg_roughness(my_core, roughness, geom_dim):
```

Comment:
Allow the user to pass in a list of vertices, and use all the vertices only if that list is empty.

Reply:
Thank you for the suggestion. I made the updates in the get_roughness function instead, so the roughness value can be that of all the vertices in the geometry or the list of vertices that user passes in.